enterprise/j2ee.ejbcore/src/org/netbeans/modules/j2ee/ejbcore/api/methodcontroller/EjbMethodController.java

```diff
@@ -72,7 +72,7 @@ public EjbType run(EjbJarMetadata metadata) throws Exception {
                 Project project = FileOwnerQuery.getOwner(ejbClassFO);
                 if (project != null){
                     J2eeProjectCapabilities projectCap = J2eeProjectCapabilities.forProject(project);
-                    allowsNoInterface = projectCap != null ? projectCap.isEjb31LiteSupported() : false;
+                    allowsNoInterface = (projectCap != null && (projectCap.isEjb31LiteSupported() || projectCap.isEjb40LiteSupported()));
```

Comment:
What's about 3.2 lite? From my understanding EJB lite is a subset of EJB full. So if EJB lite should support it as well.

Reply:
My reading of `isEjb31LiteSupported` is, that it is `true` if EJB 3.2 is supported. Assuming EJB follows the convention, that breaking changes are communicated by a change in the major number, this makes sense (repeating myself: I assume 3.2 is a superset of 3.1).
Yes if `isEjb31Supported` returns `true`, `isEjb31LiteSupported` will, the same is true for the other EJB capability checks.